src/Query/ResultPrinters/ListResultPrinter/ValueTextsBuilder.php

```diff
@@ -21,6 +23,15 @@ class ValueTextsBuilder {
 
 	private $linker;
 
+	private $mixedResults;
+
+	/**
+	 * @param SMWQueryResult $queryResult
+	 */
+	public function __construct( QueryResult $queryResult ) {
+		$this->mixedResults = !( $queryResult->getQuery()->getDescription() instanceof NamespaceDescription );
+	}
```

Comment:
If you only need the boolean, then don't inject the super complex QueryResult. That makes testing excessively difficult.

Reply:
this is the only point for which I have added a comment with further explanation within the trait src/Query/ResultPrinters/PrefixParameterProcessor